src/storage/storage.ts

```diff
@@ -78,6 +84,17 @@ export class Storage implements IStorage {
     this.version = version;
   }
 
+  setTxSign(txSign: EcdsaTxSign): void {
+    this.txSignFunc = txSign;
+  }
+
+  async signTx(tx: Transaction, pinCode: string): Promise<IInputSignature[]> {
+    if (this.txSignFunc) {
+      return this.txSignFunc(tx, this, pinCode);
+    }
+    return transactionUtils.getSignatureForTx(tx, this, pinCode);
+  }
+
```

Comment:
note: This is the core feature of this review.

Any signing on the wallet should be done from this method, this will allow us to maintain support for external signing methods going forward.

Reply:
question: Why does codecov warns us of lack of coverage here, since this is now the main form of signing transactions on the lib? Shouldn't this be covered by most of all of our transaction tests?

Reply:
The integration test does not upload the coverate artifact when it fails.

The tests are actually passing, its just the wallet facade coverate that dropped slightly but it will be back once the xpub guard PR is merged